pkg/platform/factory/factory.go

```diff
@@ -69,8 +69,11 @@ func CreatePlatform(parentLogger logger.Logger,
 		return nil, errors.Wrapf(err, "Failed to create %s platform", platformType)
 	}
 
-	if err = newPlatform.Initialize(); err != nil {
-		return nil, errors.Wrap(err, "Failed to initialize platform")
+	if platformType != "auto" {
```

Comment:
explain?

Reply:
without this "if" , `Initialize` would happen twice, once for the loop of resolving the platformType from "auto" and once after this function is called with platformType!="auto"
because this function is recursive

Reply:
oy
@liranbg  - thoughts?